### ISSUE TITLE
Missing Window Functions from Polars

### ISSUE THREAD

**Issue by guarilha** [elixir-nx/explorer#589]
Polars has the following "rolling" (explorer calls it window) functions: 

- [polars.Series.rolling_apply](https://pola-rs.github.io/polars/py-polars/html/reference/series/api/polars.Series.rolling_apply.html)
- [polars.Series.rolling_max](https://pola-rs.github.io/polars/py-polars/html/reference/series/api/polars.Series.rolling_max.html)
- [polars.Series.rolling_mean](https://pola-rs.github.io/polars/py-polars/html/reference/series/api/polars.Series.rolling_mean.html)
- [polars.Series.rolling_median](https://pola-rs.github.io/polars/py-polars/html/reference/series/api/polars.Series.rolling_median.html)
- [polars.Series.rolling_min](https://pola-rs.github.io/polars/py-polars/html/reference/series/api/polars.Series.rolling_min.html)
- [polars.Series.rolling_quantile](https://pola-rs.github.io/polars/py-polars/html/reference/series/api/polars.Series.rolling_quantile.html)
- [polars.Series.rolling_skew](https://pola-rs.github.io/polars/py-polars/html/reference/series/api/polars.Series.rolling_skew.html)
- [polars.Series.rolling_std](https://pola-rs.github.io/polars/py-polars/html/reference/series/api/polars.Series.rolling_std.html#)
- [polars.Series.rolling_sum](https://pola-rs.github.io/polars/py-polars/html/reference/series/api/polars.Series.rolling_sum.html)
- [polars.Series.rolling_var](https://pola-rs.github.io/polars/py-polars/html/reference/series/api/polars.Series.rolling_var.html)

Although some of the rolling computations are already available in Explorer, the absence of a `rolling_apply` equivalent makes it less convenient to calculate certain statistics and models. 

As a result, users are forced to resort to workarounds that are far from ideal. For example, one could calculate the 21-day rolling standard deviation by using additional columns and a combination of existing functions. However, [for users familiar with Pandas](https://pandas.pydata.org/pandas-docs/stable/user_guide/window.html), this approach can feel unusual.

Is there any plan to support `rolling_apply` in Explorer, or am I overlooking something?

**Comment by josevalim** [comment]
I don't think we can support `rolling_apply` because it is not possible to call Erlang from C/Rust without using message passing. As far as I see, the python version linked is fully implemented in C.

Can you provide a more concrete example that you are trying to address and how you are addressing it? Perhaps we can provide higher level conveniences without having it named `rolling_apply` itself?

**Comment by josevalim** [comment]
Would you like to send a PR for `Series.window_map` btw?

**Comment by guarilha** [comment]
Created this [PR](https://github.com/elixir-nx/explorer/pull/610) to explore a bit the codebase and test the waters. Waiting for review on it to make sure everything is ok. After that I plan on adding a bunch of functions that I need as well. Hope it helps.

**Comment by mrcwinn** [comment]
Hi, I also have a use case for this. Here's equivalent code in Python:

```
df['atl'] = df['tss'].rolling(window=7).apply(lambda x: calculate_atl_recursive(x))
```

Here I'm applying a calculation using a rolling window (or nil values when I'm out of historical data to cull). I can't solve this with the current package API, unless I'm missing something.

Thank you!
